src/main/Main.hs

```diff
@@ -788,6 +789,20 @@ execCmd ExecOpts {..} go@GlobalOpts{..} =
           pkgopts <- getPkgOpts menv wc pkgs
           return (prefix ++ compilerExeName wc, pkgopts ++ args)
 
+      runWithPath path callback = case path of
+        Nothing -> callback
+        Just p | not (isValid p) -> callback
+        Just p                   ->
+          if isRelative p
```

Comment:
I think it'd be cleaner to just use `System.Directory.withCurrentDirectory`.  I wish `path` had `RelOrAbs` or something, but for now `FilePath` is the best way to represent having either.